tracing-subscriber/src/fmt/time.rs

```diff
@@ -96,6 +100,109 @@ impl FormatTime for Uptime {
     }
 }
 
+/// The RFC 3339 format is used by default and using
+/// this struct allows chrono to bypass the parsing
+/// used when a custom format string is provided
+#[cfg(feature = "chrono")]
+#[derive(Debug, Clone, Eq, PartialEq)]
+enum ChronoFmtType {
+    Rfc3339,
+    Custom(String),
+}
+
+#[cfg(feature = "chrono")]
+impl Default for ChronoFmtType {
+    fn default() -> Self {
+        ChronoFmtType::Rfc3339
+    }
+}
+
+/// Retrieve and print the current UTC time.
+#[cfg(feature = "chrono")]
+#[derive(Debug, Clone, Eq, PartialEq, Default)]
+pub struct ChronoUtc {
+    format: ChronoFmtType,
+}
+
+#[cfg(feature = "chrono")]
+impl ChronoUtc {
+    /// Format the time using the [`RFC 3339`] format
+    /// (a subset of [`ISO 8601`]).
+    ///
+    /// [`RFC 3339`]: https://tools.ietf.org/html/rfc3339
+    /// [`ISO 8601`]: https://en.wikipedia.org/wiki/ISO_8601
+    pub fn new() -> Self {
+        ChronoUtc {
+            format: ChronoFmtType::Rfc3339,
+        }
+    }
+
+    /// Format the time using the given format string.
+    ///
+    /// See [`chrono::format::strftime`]
+    /// for details on the supported syntax.
+    ///
+    /// [`chrono::format::strftime`]: https://docs.rs/chrono/0.4.9/chrono/format/strftime/index.html
+    pub fn with_format(mut self, format_string: String) -> Self {
+        self.format = ChronoFmtType::Custom(format_string);
+        self
+    }
+}
+
+#[cfg(feature = "chrono")]
+impl FormatTime for ChronoUtc {
+    fn format_time(&self, w: &mut dyn fmt::Write) -> fmt::Result {
+        let time = chrono::Utc::now();
+        match self.format {
+            ChronoFmtType::Rfc3339 => write!(w, "{} ", time.to_rfc3339()),
+            ChronoFmtType::Custom(ref format_str) => write!(w, "{} ", time.format(format_str)),
+        }
+    }
+}
+
+/// Retrieve and print the current local time.
+#[cfg(feature = "chrono")]
+#[derive(Debug, Clone, Eq, PartialEq, Default)]
+pub struct ChronoLocal {
+    format: ChronoFmtType,
+}
+
+#[cfg(feature = "chrono")]
+impl ChronoLocal {
+    /// Format the time using the [`RFC 3339`] format
```

Comment:
WDYT about having separate `with_format` and `rfc_3339` constructors, instead of defaulting to RFC 3339 and then allowing the formatter to be mutated to use a custom format? This seems surprising, and I'd prefer the constructor for the RFC3339 variant to explicitly name the formatting rather than just being "new"...

Reply:
Implemented in https://github.com/tokio-rs/tracing/pull/387/commits/a18e5c9f2f3e7cc5b4fcca49850e5a4877fa9948